lib/plug/router.ex

```diff
@@ -121,9 +121,9 @@ defmodule Plug.Router do
 
   ## Routes compilation
 
-  All routes are compiled to a match function that receives
-  three arguments: the method, the request path split on `/`
-  and the connection. Consider this example:
+  All routes are compiled to a do_match function that receives
```

Comment:
> did you mean that we fake a compiled result function that would be understandable by users reading the doc?

Yes, I think so. We still need to make it clear that both match and dispatch function are defined and give an idea of how dispatch works.